Native-image-maven-plugin does not work in Windows

Hello, I am having trouble while compiling springboot using maven v3.1.1, the error message is
 

`Fatal error: com.oracle.svm.core.util.VMError$HostedError: Option name "IncludeResourceBundles" has multiple definitions: com.oracle.svm.core.jdk.LocalizationSupport$Options.IncludeResourceBundles and com.oracle.svm.core.jdk.LocalizationFeature$Options.IncludeResourceBundles
	at com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:66)
	at com.oracle.svm.hosted.option.HostedOptionParser.collectOptions(HostedOptionParser.java:88)
	at com.oracle.svm.hosted.option.HostedOptionParser.<init>(HostedOptionParser.java:61)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:206)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:454)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:115)`


Can someone help me ?

Thank you for your answer. But I have another problem, I have already followed the steps of issues / 1866. But to no avail :(   

`[ERROR] Failed to execute goal org.graalvm.nativeimage:native-image-maven-plugin:19.3.1:native-image (default) on project s1-server-pdv-mono-application: Image building on Java 11+ without native-image requires MAVEN_OPTS='--add-exports=java.base/jdk.internal.module=ALL-UNNAMED' -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.graalvm.nativeimage:native-image-maven-plugin:19.3.1:native-image (default) on project s1-server-pdv-mono-application: Image building on Java 11+ without native-image requires MAVEN_OPTS='--add-exports=java.base/jdk.internal.module=ALL-UNNAMED'
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Image building on Java 11+ without native-image requires MAVEN_OPTS='--add-exports=java.base/jdk.internal.module=ALL-UNNAMED'
    at com.oracle.substratevm.NativeImageMojo.execute (NativeImageMojo.java:255)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException`


**when trying to create native image**
With command - > cd %JAVA_HOME% REM Change to GraalVM root dir
bin\native-image.cmd -jar lib\graalvm\svm-driver.jar -H:Path=.\bin REM Build yourself a native-image.exe next to bin\native-image.cmd

` Creating library C:\graalvm-ee-java11-19.3.1\native-image.lib and object C:\graalvm-ee-java11-19.3.1\native-image.exp
net.lib(NetworkInterface.obj) : error LNK2019: unresolved external symbol __imp___stdio_common_vsnprintf_s referenced in function _snprintf_s
net.lib(NetworkInterface_winXP.obj) : error LNK2001: unresolved external symbol __imp___stdio_common_vsnprintf_s
net.lib(net_util_md.obj) : error LNK2019: unresolved external symbol __imp___acrt_iob_func referenced in function printf
java.lib(jni_util.obj) : error LNK2001: unresolved external symbol __imp___acrt_iob_func
net.lib(net_util_md.obj) : error LNK2019: unresolved external symbol __imp___stdio_common_vfprintf referenced in function _vfprintf_l
java.lib(jni_util.obj) : error LNK2001: unresolved external symbol __imp___stdio_common_vfprintf
jvm.lib(JvmFuncs.obj) : error LNK2001: unresolved external symbol __imp___stdio_common_vfprintf
net.lib(net_util_md.obj) : error LNK2019: unresolved external symbol __imp___stdio_common_vsprintf referenced in function _vsnprintf_l
net.lib(NetworkInterface_winXP.obj) : error LNK2001: unresolved external symbol __imp___stdio_common_vsprintf
jvm.lib(JvmFuncs.obj) : error LNK2001: unresolved external symbol __imp___stdio_common_vsprintf
java.lib(ProcessImpl_md.obj) : error LNK2019: unresolved external symbol __report_rangecheckfailure referenced in function win32Error
zip.lib(inftrees.obj) : error LNK2001: unresolved external symbol __report_rangecheckfailure
java.lib(ProcessImpl_md.obj) : error LNK2019: unresolved external symbol __imp___stdio_common_vswprintf referenced in function _vswprintf_c_l
C:\graalvm-ee-java11-19.3.1\native-image.exe : fatal error LNK1120: 6 unresolved externals

        at com.oracle.svm.hosted.image.NativeBootImageViaCC.write(NativeBootImageViaCC.java:388)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:652)
        at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:445)
        at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Error: Image build request failed with exit status 1`

Sorry, I'm sure I have native-image installed as I conclude from this message when I run "gu install native-image".

   Downloading: Component catalog from www.graalvm.org
   Processing Component: Native Image
   Component Native Image (org.graalvm.native-image) is already installed.

graalvm version is

   openjdk version "11.0.7" 2020-04-14
   OpenJDK Runtime Environment GraalVM CE 20.1.0 (build 11.0.7+10-jvmci-20.1-b02)
   OpenJDK 64-Bit Server VM GraalVM CE 20.1.0 (build 11.0.7+10-jvmci-20.1-b02, mixed mode, sharing)

I'm using the corresponding maven plugin

   [ERROR] Failed to execute goal org.graalvm.nativeimage:native-image-maven-plugin:20.1.0:native-image (default) on project ruu.io-graalvm-sandbox: Image building on Java 11+ without native-image requires MAVEN_OPTS='--add-exports=java.base/jdk.internal.module=ALL-UNNAMED

Any ideas? Thanx.

--- a/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/OS.java
+++ b/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/OS.java
@@ -52,6 +52,10 @@ public enum OS {
         return name().toLowerCase();
     }
 
+    public boolean isCurrent() {
+        return getCurrent() == this;
+    }
+
     private static OS findCurrent() {
         final String name = System.getProperty("os.name");
         if (name.equals("Linux")) {

--- a/substratevm/src/native-image-maven-plugin/src/main/java/com/oracle/substratevm/NativeImageMojo.java
+++ b/substratevm/src/native-image-maven-plugin/src/main/java/com/oracle/substratevm/NativeImageMojo.java
@@ -175,7 +175,7 @@ public class NativeImageMojo extends AbstractMojo {
         addClasspath(project.getArtifact());
         String classpathStr = classpath.stream().map(Path::toString).collect(Collectors.joining(File.pathSeparator));
 
-        Path nativeImageExecutable = getMojoJavaHome().resolve("bin").resolve(withExeSuffix("native-image"));
+        Path nativeImageExecutable = getMojoJavaHome().resolve("bin").resolve("native-image" + (OS.WINDOWS.isCurrent() ? ".cmd" : ""));
         if (Files.isExecutable(nativeImageExecutable)) {
             String nativeImageExecutableVersion = "Unknown";
             Process versionCheckProcess = null;
@@ -266,13 +266,6 @@ public class NativeImageMojo extends AbstractMojo {
                 throw new MojoExecutionException("Image building on Java 11+ without native-image requires MAVEN_OPTS='--add-exports=java.base/jdk.internal.module=ALL-UNNAMED'", e);
             }
         }
-    }
-
-    private String withExeSuffix(String basename) {
-        if (OS.getCurrent() == OS.WINDOWS) {
-            return basename + ".exe";
-        }
-        return basename;
     }
 
     private void addClasspath(Artifact artifact) throws MojoExecutionException {
@@ -421,7 +414,7 @@ public class NativeImageMojo extends AbstractMojo {
 
         @Override
         public Path getJavaExecutable() {
-            return getJavaHome().resolve("bin").resolve(withExeSuffix("java"));
+            return getJavaHome().resolve("bin").resolve("java" + (OS.WINDOWS.isCurrent() ? ".exe" : ""));
         }
 
         private List<Path> getSelectedArtifactPaths(String groupId, String... artifactIds) {

